Content.Server/Physics/Controllers/MoverController.cs

```diff
@@ -63,46 +128,19 @@ protected override bool CanSound()
         return true;
     }
 
-    private HashSet<EntityUid> _moverAdded = new();
-    private List<Entity<InputMoverComponent>> _movers = new();
-
-    private void InsertMover(Entity<InputMoverComponent> source)
-    {
-        if (TryComp(source, out MovementRelayTargetComponent? relay))
-        {
-            if (TryComp(relay.Source, out InputMoverComponent? relayMover))
-            {
-                InsertMover((relay.Source, relayMover));
-            }
-        }
-
-        // Already added
-        if (!_moverAdded.Add(source.Owner))
-            return;
-
-        _movers.Add(source);
-    }
-
     public override void UpdateBeforeSolve(bool prediction, float frameTime)
     {
         base.UpdateBeforeSolve(prediction, frameTime);
 
-        _moverAdded.Clear();
-        _movers.Clear();
-        var inputQueryEnumerator = AllEntityQuery<InputMoverComponent>();
-
-        // Need to order mob movement so that movers don't run before their relays.
-        while (inputQueryEnumerator.MoveNext(out var uid, out var mover))
-        {
-            InsertMover((uid, mover));
-        }
+        var inputQueryEnumerator = AllEntityQuery<ActiveInputMoverComponent, InputMoverComponent>();
 
-        foreach (var mover in _movers)
+        var count = 0;
+        while (inputQueryEnumerator.MoveNext(out var uid, out _, out var moverComp))
         {
-            HandleMobMovement(mover, frameTime);
+            count++;
+            HandleMobMovement((uid, moverComp), frameTime);
         }
```

Comment:
Is there a reason why the ordering fix was removed? This fixed relays running before their parents but removing that turns it into the wild-west again and adds in entity-ordering bugs.

Reply:
Nope I just forgot to bring that back over; thanks!